### PR TITLE
Configure an IBM HTTP Server in the tWAS cluster

### LINKED ISSUE #37: Modify ARM template to stand up VM based on IHS, use as Load Balancer
Depends on [gh-image-28](https://github.com/WASdev/azure.websphere-traditional.image/issues/28).

### LINKED ISSUE #40: Enhance text in `createUiDefinition` to aid people getting an entitlement.
The current text in the `createUiDefinition.json` regarding entitlement says:

> Provide an entitled IBMid to provision the WebSphere cluster. [Learn more](https://www.ibm.com/ibmid/myibm/help/us/helpdesk.html).

This issue requests that we enhance the text:
* To make it very clear that the deployment will fail if the IBMid is not entitled.
* Give the user some guidance on how to go about getting an entitlement.

### LINKED ISSUE #84: Use "IBM ID" rather than IBMid consistently
This requirement is from the [comment](https://github.com/WASdev/azure.websphere-traditional.cluster/issues/35#issuecomment-874015065) of #35 :
> Minor suggestion and maybe we need a separate issue for it: can we use "IBM ID" (rather than IBMid) consistently as I think this is the correct/official representation?

### PR DESCRIPTION
**Pull request by majguo**
## Description
This PR is to resolve the following issues:
* Resolve #37 
* Resolve #40 
* Resolve #84 

## Change summary
* Automation scripts to configure IHS on both ihs & dmgr VMs
* ARM template to deploy Azure resources for IHS 
* UI to configure IHS on Azure Portal 
* Enhance text in `createUiDefinition.json` to aid people getting an entitlement
* Use IBM ID rather than IBMid consistently

## Test cases
The following cases have already been verified before sending out the PR. You can have a try using the [preview link](https://portal.azure.com/#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewwasndt91-n-cluster) also.
* Successfully deployed a tWAS cluster with an a working IHS if specified by the user on Azure Portal;
  * Successfully installed `DefaultApplication` (shipped with WAS ND installation) which can be visited from the public IP address of IHS;
* Successfully deployed a tWAS cluster without an an IHS if not specified by the user on Azure Portal;

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: src/main/arm/createUiDefinition.json [hunk 1]
```diff
@@ -9,28 +9,28 @@
                 "type": "Microsoft.Common.InfoBox",
                 "options": {
                     "icon": "Info",
-                    "text": "Provide an entitled IBMid to provision the WebSphere cluster. <a href='https://www.ibm.com/ibmid/myibm/help/us/helpdesk.html' target='_blank'>Learn more</a>"
+                    "text": "This offer is Bring-Your-Own-License. You will be required to enter your registered IBM ID prior to successfully deploying this offer and your IBM ID must have active WebSphere entitlements associated with it. If you find that provisioning fails due to lack of entitlements, please contact the primary or secondary contacts for your IBM Passport Advantage Site who should be able to grant you access or follow steps at <a href='https://www-112.ibm.com/software/howtobuy/passportadvantage/homepage/ecarec' target='_blank'>IBM eCustomer Care</a> for further assistance. This offer also assumes you are properly licensed to run offers in Microsoft Azure."
```
Comment (git4rk) [inline src/main/arm/createUiDefinition.json:12]: I have requested our documentation team to review this message.

Also, I am not sure what this means: `This offer also assumes you are properly licensed to run offers in Microsoft Azure.`? How can a customer check that?

File: src/main/arm/createUiDefinition.json [hunk 2]
```diff
@@ -237,6 +226,153 @@
...
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "IHS configuration",
+                "elements": [
+                    {
+                        "name": "configIHS",
+                        "type": "Microsoft.Common.OptionsGroup",
+                        "label": "Configure an IBM HTTP Server?",
+                        "defaultValue": "Yes",
+                        "toolTip": "Select 'Yes' and provide required info to configure an IBM HTTP Server.",
```
Comment (git4rk) [inline src/main/arm/createUiDefinition.json:244]: I think we should mention that if they select `No` they can not use this offer to add an HTTP server later. They will either need to set up a new cluster or add an HTTP Server manually.

File: src/main/arm/createUiDefinition.json (hunk 1 above)
Comment (majguo) [reply src/main/arm/createUiDefinition.json:12]: @git4rk  I copied message from #40, which is provided by @gcharters . Ranjan, could you pls work with Graham to figure out the final version of the message?

File: src/main/arm/createUiDefinition.json (hunk 2 above)
Comment (majguo) [reply src/main/arm/createUiDefinition.json:244]: Good point. @git4rk Could you pls provide the improved tool tip message based on your thoughts above? Maybe you can also request your documentation team to review this message.

File: src/main/arm/createUiDefinition.json (hunk 1 above)
Comment (git4rk) [reply src/main/arm/createUiDefinition.json:12]: @majguo Here is the message from our documentation team:

This offer is Bring-Your-Own-License. To deploy this offer, you must enter your registered IBMid and your IBMid must have active WebSphere entitlements associated with it. If provisioning fails due to lack of entitlements, ask the primary or secondary contacts for your IBM Passport Advantage site to grant you access or follow steps at [IBM eCustomer Care](https://www-112.ibm.com/software/howtobuy/passportadvantage/homepage/ecarec) for further assistance. This offer also assumes you are properly licensed to run offers in Microsoft Azure.

File: src/main/arm/createUiDefinition.json (hunk 2 above)
Comment (git4rk) [reply src/main/arm/createUiDefinition.json:244]: @majguo Here is the suggested text (after reviewing from doc team):
Select 'Yes' and provide the required information to configure an IBM HTTP Server. If you select 'No', you will not be able to use this offer to add IBM HTTP Server later to an existing cluster. You either must set up a new cluster or add IBM HTTP Server manually.

File: src/main/scripts/configure-ihs-on-dmgr.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+# Mount Azure File Share system
+mkdir -p $mountpointPath
+mkdir /etc/smbcredentials
+echo "username=$storageAccountName" > /etc/smbcredentials/${storageAccountName}.cred
+echo "password=$storageAccountKey" >> /etc/smbcredentials/${storageAccountName}.cred
+chmod 600 /etc/smbcredentials/${storageAccountName}.cred
+echo "//${storageAccountName}.file.core.windows.net/${fileShareName} $mountpointPath cifs nofail,vers=2.1,credentials=/etc/smbcredentials/${storageAccountName}.cred,dir_mode=0777,file_mode=0777,serverino" >> /etc/fstab
+
+yum install cifs-utils -y
```
Comment (git4rk) [inline src/main/scripts/configure-ihs-on-dmgr.sh:41]: Can we do this on the image itself?

Comment (majguo) [reply src/main/scripts/configure-ihs-on-dmgr.sh:41]: Yes, we can. The reason why it's moved to the image is that IHS configuration is an optional step. If user doesn't configure an IHS using the offer but manually set it up for some reason, he/she can use other approaches to transfer configuration script between IHS and DMgr VMs.

File: src/main/arm/createUiDefinition.json (hunk 1 above)
Comment (majguo) [reply src/main/arm/createUiDefinition.json]: Thanks. Fixed in b6c4aa8.

File: src/main/arm/createUiDefinition.json (hunk 2 above)
Comment (majguo) [reply src/main/arm/createUiDefinition.json]: Thanks. Fixed in b6c4aa8.

File: src/main/scripts/configure-ihs-on-dmgr.sh (hunk 3 above)
Comment (majguo) [reply src/main/scripts/configure-ihs-on-dmgr.sh:41]: @git4rk Let's approach in this way:
1. Keep `yum install cifs-utils -y` now
1. Update cicd pipelines for both twas-nd and ihs base images to install cifs-utils preparing VM
1. Remove `yum install cifs-utils -y` from deployment scripts in cluster repo

I opened xxx to track the

**Review by git4rk — APPROVED** [review]

**Pull request merged**